src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -1002,54 +1002,77 @@ RandomBlockOolWriter::do_write(
-
     ex->prepare_write();
-    extent_len_t offset = 0;
+
+    RandomBlockManager * rbm = rb_cleaner->get_rbm(ex->get_paddr()); 
+    assert(rbm);
+    paddr_t cur_base = ex->get_paddr();
+    assert(ex_base.is_absolute());
+    extent_len_t cur_len = ex->get_length();
     bufferptr bp;
+    TRACE("write extent {}, paddr {} ...",
+	  fmt::ptr(ex.get()), cur_base);
     if (can_inplace_rewrite(t, ex)) {
       assert(ex->is_logical());
       auto r = ex->template cast<LogicalCachedExtent>()->get_modified_region();
       ceph_assert(r.has_value());
-      offset = p2align(r->offset, rbm->get_block_size());
-      extent_len_t len =
-	p2roundup(r->offset + r->len, rbm->get_block_size()) - offset;
-      bp = ceph::bufferptr(ex->get_bptr(), offset, len);
+      extent_len_t offset = p2align(r->offset, rbm->get_block_size());
+      cur_len = p2roundup(r->offset + r->len, rbm->get_block_size()) - offset;
+      cur_base = ex->get_paddr() + offset;
+      bp = ceph::bufferptr(ex->get_bptr(), offset, cur_len);
     } else {
       bp = ex->get_bptr();
       auto& trans_stats = get_by_src(w_stats.stats_by_src, t.get_src());
-      ++(trans_stats.num_records);
       trans_stats.data_bytes += ex->get_length();
       w_stats.data_bytes += ex->get_length();
     }
-    writes.push_back(write_info_t{paddr + offset, std::move(bp), rbm});
 
     if (ex->is_initial_pending()) {
       t.mark_allocated_extent_ool(ex);
     } else if (can_inplace_rewrite(t, ex)) {
       assert(ex->is_logical());
       t.mark_inplace_rewrite_extent_ool(
-        ex->template cast<LogicalCachedExtent>());
+	ex->template cast<LogicalCachedExtent>());
     } else {
       ceph_assert("impossible");
     }
+
+    // TODO : allocate a consecutive address based on a transaction
+    if (writes.size() != 0 &&
+       (writes.back().offset + writes.back().len == cur_base &&
+       writes.back().bp.c_str() + writes.back().len == ex->get_bptr().c_str())) {
+      // We can write both the currrent extent and the previous one at once
+      // if the extents are located in a raw
+      writes.back().len += cur_len;
+      writes.back().bp = buffer::claim_char(
+	writes.back().len, writes.back().bp.c_str());
+    } else {
+      // Write a single extent in the existing way
+      write_info_t w_info;
+      w_info.offset = cur_base;
+      w_info.rbm = rbm;
+      w_info.len = cur_len;
+      w_info.bp = bp;
+      writes.push_back(w_info);
+    }
+    TRACE("current extent: base off {} len {}, previous extent: base off {} len {}",
+      cur_base, cur_len, writes.back().offset, writes.back().len);
   }
 
   return trans_intr::make_interruptible(
     seastar::do_with(std::move(writes),
-      [](auto& writes) {
+      [&t, this](auto& writes) {
       return crimson::do_for_each(writes,
-        [](auto& info) {
+        [&t, this](auto& info) {
+	auto& stats = t.get_ool_write_stats();
+	stats.num_records += 1;
+	auto& trans_stats = get_by_src(w_stats.stats_by_src, t.get_src());
```

Comment:
Nit, move stats references out of `do_for_each`, there is no need to find the stats in each repeat.

Reply:
Done.